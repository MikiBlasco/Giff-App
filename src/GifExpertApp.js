import React from 'react';
import { useState } from 'react';
import AddCategory from './Components/AddCategory';
import { GiffGrid } from './Components/GiffGrid';


const GifExpertApp = ( { defaultCategories = [] } ) => {

    //const [categories, setCategories] = useState(['Dragon Ball']);
    const [categories, setCategories] = useState( defaultCategories );

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

          <ol>
              { categories.map( (category) =>


               <GiffGrid  key={category} category={category} />

    


              )  }
          </ol>

        </>
    )

}

export default GifExpertApp