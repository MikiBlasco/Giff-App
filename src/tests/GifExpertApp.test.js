import React from 'react';
import '@testing-library/jest-dom';
import  {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';



describe('pruebas del componente <GifExpertApp/>', () => {

   test('should match with snapshot', () => { 
       const wrapper = shallow(<GifExpertApp/>)
       expect(wrapper).toMatchSnapshot();
    })

    test('should show a list of categories', () => { 

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/> );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiffGrid').length).toBe(categories.length);
        
     })

})