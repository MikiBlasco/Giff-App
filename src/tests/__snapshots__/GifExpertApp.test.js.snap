// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del componente <GifExpertApp/> should match with snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas del componente <GifExpertApp/> should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiffGrid
      category="One Punch"
      key="One Punch"
    />
    <GiffGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
